src/components/JobsContext.jsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import { JOBS_DOCTYPE } from '../doctypes'
+import CozyRealtime from 'cozy-realtime'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import logger from 'cozy-logger'
+import { useI18n } from 'cozy-ui/react/I18n'
+
+const log = logger.namespace('import.context')
+
+export const JobsContext = createContext({})
+
+export const useJobsContext = () => {
+  return useContext(JobsContext)
+}
+
+const JobsProvider = ({ children, client }) => {
+  const [jobsInProgress, setJobsInProgress] = useState([])
+  const { t } = useI18n()
+  const realtimeStartedRef = useRef(false)
+  const jobsInProgressRef = useRef([])
+
+  const realtime = useMemo(() => {
+    return new CozyRealtime({ client })
+  }, [client])
+
+  const handleRealtime = data => {
+    const { worker, state, message: msg } = data
+    const { current: currJobsInProgress } = jobsInProgressRef
+    const index = currJobsInProgress.findIndex(a => a.account === msg.account)
+    const exist = index !== -1
+
+    let arr = [...currJobsInProgress]
+    if (worker === 'konnector') {
+      if (state === 'running' && !exist) {
+        arr.push(msg)
+      } else if (state === 'done' && exist) {
+        arr.splice(index, 1)
+        Alerter.success(t('JobsContext.alerter-success'))
+      } else if (state === 'errored' && exist) {
+        arr.splice(index, 1)
+        Alerter.error(t('JobsContext.alerter-errored'))
+      }
+      jobsInProgressRef.current = arr
+      setJobsInProgress(jobsInProgressRef.current)
+    }
+  }
+
+  const startJobsRealtime = () => {
+    if (!realtimeStartedRef.current) {
+      log('info', 'Start Jobs Realtime')
+      realtimeStartedRef.current = true
+
+      realtime.subscribe('created', JOBS_DOCTYPE, handleRealtime)
+      realtime.subscribe('updated', JOBS_DOCTYPE, handleRealtime)
+      realtime.subscribe('deleted', JOBS_DOCTYPE, handleRealtime)
+    }
+  }
+
+  const stopJobsRealtime = () => {
+    if (realtimeStartedRef.current) {
+      log('info', 'Stop Jobs Realtime')
+      realtimeStartedRef.current = false
+
+      realtime.unsubscribe('created', JOBS_DOCTYPE, handleRealtime)
+      realtime.unsubscribe('updated', JOBS_DOCTYPE, handleRealtime)
+      realtime.unsubscribe('deleted', JOBS_DOCTYPE, handleRealtime)
+    }
+  }
+
+  useEffect(() => {
+    startJobsRealtime()
+    return () => stopJobsRealtime()
+  }, [])
```

Comment:
There is useRealtime hook that could simplify the logic, it does the start/stop/subscribe/unsubscribe for you:

https://github.com/cozy/cozy-ui/blob/db54f27493b6ba61cf41a1b48cf363b9957dd049/react/hooks/useRealtime.js

cozy-ui might not be the good place for it though.

Reply:
Yes I think so too, it would be better in realtime lib